### PR TITLE
Fix/warnings

### PR DESCRIPTION
**Pull request by hroest**
fix a few warnings by clang

### CODE REVIEW

**Comment by hroest** [comment]
see also #757 for a collection of a few warnings

**Comment by hroest** [comment]
this should bring down the number of warnings with clang to 174 (5 in the headers)

- for the library only
```
      1 [-Woverloaded-virtual]
      1 [-Wsign-compare]
      1 [-Wuninitialized]
      1 [-Wunknown-pragmas]
      1 [-Wunused-parameter]
      2 [-Wunused-macros]
      3 [-Wc++11-extensions]
      3 [-Wunused-function]
      5 [-Wcovered-switch-default]
      5 [-Wmissing-noreturn]
      6 [-Wconditional-uninitialized]
      6 [-Wunreachable-code]
     41 [-Wshadow]
```

- for all code
```
      1 [-Wlanguage-extension-token]
      1 [-Woverloaded-virtual]
      1 [-Wsign-compare]
      1 [-Wuninitialized]
      1 [-Wunknown-pragmas]
      1 [-Wunused-parameter]
      1 [-Wvariadic-macros]
      2 [-Wheader-hygiene]
      2 [-Wunused-exception-parameter]
      2 [-Wunused-macros]
      3 [-Wc++11-extensions]
      3 [-Wunused-function]
      5 [-Wcovered-switch-default]
      6 [-Wconditional-uninitialized]
      7 [-Wmissing-noreturn]
      7 [-Wunreachable-code]
    130 [-Wshadow]
```

with only these files generating more then 2 warnings

```
      3 src/openms/source/ANALYSIS/ID/PILISScoring.cpp
      3 src/openms/source/ANALYSIS/TARGETED/PSLPFormulation.cpp
      3 src/openms/source/TRANSFORMATIONS/RAW2PEAK/PeakPickerCWT.cpp
      3 src/tests/class_tests/openms/source/ClassTest_test.cpp
      3 src/tests/class_tests/openms/source/ConsensusFeature_test.cpp
      3 src/topp/FileFilter.cpp
      4 src/openms/source/ANALYSIS/MAPMATCHING/PoseClusteringAffineSuperimposer.cpp
      4 src/openms/source/ANALYSIS/TARGETED/PrecursorIonSelection.cpp
      4 src/openms/source/TRANSFORMATIONS/RAW2PEAK/TwoDOptimization.cpp
      4 src/tests/class_tests/openms/source/Matrix_test.cpp
      4 src/tests/class_tests/openms/source/Param_test.cpp
      4 src/utils/OpenSwathWorkflow.cpp
      7 src/openms/source/ANALYSIS/ID/ProtonDistributionModel.cpp
```

File: src/openms/include/OpenMS/TRANSFORMATIONS/FEATUREFINDER/IsotopeWaveletTransform.h [hunk 1]
```diff
@@ -527,7 +527,7 @@ namespace OpenMS
     Int from_max_to_left_, from_max_to_right_;
     std::vector<int> indices_;
 
-    MSSpectrum<PeakType> c_sorted_candidate_;
+    // MSSpectrum<PeakType> c_sorted_candidate_; // TODO this seems unused
```
Comment (cbielow) [inline src/openms/include/OpenMS/TRANSFORMATIONS/FEATUREFINDER/IsotopeWaveletTransform.h]: then delete it?! :)

File: src/openms/source/TRANSFORMATIONS/FEATUREFINDER/IsotopeWavelet.cpp [hunk 2]
```diff
@@ -48,8 +48,10 @@
 #define TWOPI 6.283185307
 #endif
 
+// TODO why should this define guard be needed ?
 #ifndef SINETRAINVECS
```
Comment (cbielow) [inline src/openms/source/TRANSFORMATIONS/FEATUREFINDER/IsotopeWavelet.cpp]: remove

File: src/openms/source/TRANSFORMATIONS/RAW2PEAK/PeakShape.cpp [hunk 3]
```diff
@@ -53,31 +53,32 @@ namespace OpenMS
...
 
-  PeakShape::PeakShape(double height, double mz_position, double left_width, double right_width, double area, PeakIterator left, PeakIterator right, Type type) :
-    height(height),
-    mz_position(mz_position),
-    left_width(left_width),
-    right_width(right_width),
-    area(area),
+  PeakShape::PeakShape(double height_, double mz_position_, double left_width_, 
+      double right_width_, double area_, PeakIterator left_, PeakIterator right_, Type type_) :
+    height(height_),
```
Comment (cbielow) [inline src/openms/source/TRANSFORMATIONS/RAW2PEAK/PeakShape.cpp]: not sure about the underscore .. my first thought was 'member?'
maybe XY_rhs?

File: src/tests/class_tests/openms/source/InclusionExclusionList_test.cpp [hunk 4]
```diff
@@ -45,6 +45,9 @@
 using namespace OpenMS;
 using namespace std;
 
+#pragma clang diagnostic push
+#pragma clang diagnostic ignored "-Wshadow"
```
Comment (cbielow) [inline src/tests/class_tests/openms/source/InclusionExclusionList_test.cpp]: is actively ignoring the warning helping in any way?

File: src/topp/OpenSwathChromatogramExtractor.cpp [hunk 5]
```diff
@@ -52,11 +52,11 @@ using namespace std;
 #include <omp.h>
 #endif
 
-#ifdef _OPENMP
-  #define IF_MASTERTHREAD if (omp_get_thread_num() ==0)  
-#else
-  #define IF_MASTERTHREAD 
-#endif    
+// #ifdef _OPENMP
```
Comment (cbielow) [inline src/topp/OpenSwathChromatogramExtractor.cpp]: remove?!
The code using it is commented out anyways (and seems wrong anyways -- so delete as well?)

File: src/openms/source/TRANSFORMATIONS/RAW2PEAK/PeakShape.cpp (hunk 3 above)
Comment (hroest) [inline src/openms/source/TRANSFORMATIONS/RAW2PEAK/PeakShape.cpp]: I used XY_arg

strangely, the members are public and therefore dont have a  trailing underscore

File: src/tests/class_tests/openms/source/InclusionExclusionList_test.cpp (hunk 4 above)
Comment (hroest) [inline src/tests/class_tests/openms/source/InclusionExclusionList_test.cpp]: no, but for the test I actually dont really care - however, it pollutes the warnings in cdash

File: src/topp/OpenSwathChromatogramExtractor.cpp (hunk 5 above)
Comment (hroest) [inline src/topp/OpenSwathChromatogramExtractor.cpp]: the code is correct and useful for debugging purposes

Comment (cbielow) [inline src/topp/OpenSwathChromatogramExtractor.cpp]: I know, its copied from MSSim :)

but the place where its was used is commented out, so why keep either of 
them?

Am 4/22/2015 um 19:16 schrieb Hannes Roest:

**Comment by cbielow** [comment]
looks good, thx.
Will merge.

**Pull request merged**